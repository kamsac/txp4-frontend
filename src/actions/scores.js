import dotenvConfiguration from '../dotenv-configuration';
import ScoresResourceMock from '../resources/scores/mock';
import ScoresResource from '../resources/scores';

export const REQUEST_SCORES = 'scores:requestScores';
export function requestScores() {
  return dispatch => {
    return dispatch(fetchScores());
  }
}

function fetchScores() {
  const resource = dotenvConfiguration.API_URL ? ScoresResource : ScoresResourceMock;

  return dispatch => {
    resource.getScores()
      .then((response) => {
        dispatch(receiveScores(response.data));
      });
  }
}

export const RECEIVE_SCORES = 'scores:fetchScoresSuccess';
export function receiveScores(scores) {
  return {
    type: RECEIVE_SCORES,
    payload: { scores },
  }
}
