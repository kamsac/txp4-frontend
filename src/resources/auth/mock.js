export const ACCESS_TOKEN = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiaWF0IjoxNTE2MjM5MDIyLCJsb2dpbiI6Imx5bWFrIiwibmljayI6IiRzJG8kYTYwbCRjODJ5JGZhM20kYzgyYSRhNjBrIn0.aaKzXEZP2tovhBAqPJYBdmbLzKjfdBsuX3FaB58JZvE';

const AuthResourceMock = {
  login() {
    return new Promise((resolve) => {
      const response = {
        data: {
          token: ACCESS_TOKEN,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default AuthResourceMock;
