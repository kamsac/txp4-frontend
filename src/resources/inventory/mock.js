const ITEMS = [
  {
    id: "a1a1a1",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: true,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "b2b2b2",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "c3c3c3",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "d4d4d4",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "e5e5e5",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "f6f6f6",
    name: "Duper engine",
    vendorId: 'aaa',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1.5,
      },
      {
        name: "acceleration",
        value: 3,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "f7f7f7",
    name: "Tire",
    vendorId: 'aaa',
    equipRegion: "tire",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1.2,
      },
      {
        name: "steering",
        value: 1.1,
      },
      {
        name: "breaking",
        value: 1.2,
      },
      {
        name: "grip",
        value: 1.5,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "g8g8g8",
    name: "Less tired",
    vendorId: 'bbb',
    equipRegion: "tire",
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.4,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1.4,
      },
      {
        name: "breaking",
        value: 2,
      },
      {
        name: "grip",
        value: 3,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
  {
    id: "h9h9h9",
    name: "Shit engine",
    vendorId: 'ccc',
    equipRegion: "engine",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 2,
      },
      {
        name: "acceleration",
        value: 1.8,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 0.2,
      },
      {
        name: "gravity",
        value: 1.5,
      },
    ],
  },
  {
    id: "i0i0i0",
    name: "Some tire",
    vendorId: 'ddd',
    equipRegion: "tire",
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 1,
      },
      {
        name: "acceleration",
        value: 1,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1.1,
      },
      {
        name: "grip",
        value: 0.8,
      },
      {
        name: "gravity",
        value: 0.9,
      },
    ],
  },
  {
    id: "j1j1j1",
    name: "Transmission",
    vendorId: 'ccc',
    equipRegion: "transmission",
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: "maxSpeed",
        value: 0.7,
      },
      {
        name: "acceleration",
        value: 1.5,
      },
      {
        name: "steering",
        value: 1,
      },
      {
        name: "breaking",
        value: 1,
      },
      {
        name: "grip",
        value: 1,
      },
      {
        name: "gravity",
        value: 1,
      },
    ],
  },
];

const InventoryResourceMock = {
  getItems() {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  equipItem(item) {
    return new Promise((resolve, reject) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(ITEMS), 100);
    });
  },
};

export default InventoryResourceMock;
