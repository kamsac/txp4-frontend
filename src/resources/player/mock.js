export const ITEMS = [
  {
    id: 'a1a1a1',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: true,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'b2b2b2',
    name: 'Less tired',
    vendorId: 'bbb',
    equipRegion: 'tire',
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.4,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1.4,
      },
      {
        name: 'breaking',
        value: 2,
      },
      {
        name: 'grip',
        value: 3,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'c3c3c3',
    name: 'Shit engine',
    vendorId: 'ccc',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 2,
      },
      {
        name: 'acceleration',
        value: 1.8,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 0.2,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'd4d4d4',
    name: 'Some tire',
    vendorId: 'ddd',
    equipRegion: 'tire',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1.1,
      },
      {
        name: 'grip',
        value: 0.8,
      },
      {
        name: 'gravity',
        value: 0.9,
      },
    ],
  },
  {
    id: 'e5e5e5',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'f6f6f6',
    name: 'Duper engine',
    vendorId: 'aaa',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1.5,
      },
      {
        name: 'acceleration',
        value: 3,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 1,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'f7f7f7',
    name: 'Tire',
    vendorId: 'aaa',
    equipRegion: 'tire',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1.2,
      },
      {
        name: 'steering',
        value: 1.1,
      },
      {
        name: 'breaking',
        value: 1.2,
      },
      {
        name: 'grip',
        value: 1.5,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'g8g8g8',
    name: 'Less tired',
    vendorId: 'bbb',
    equipRegion: 'tire',
    tier: 1,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.4,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1.4,
      },
      {
        name: 'breaking',
        value: 2,
      },
      {
        name: 'grip',
        value: 3,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
  {
    id: 'h9h9h9',
    name: 'Shit engine',
    vendorId: 'ccc',
    equipRegion: 'engine',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 2,
      },
      {
        name: 'acceleration',
        value: 1.8,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 0.2,
      },
      {
        name: 'gravity',
        value: 1.5,
      },
    ],
  },
  {
    id: 'i0i0i0',
    name: 'Some tire',
    vendorId: 'ddd',
    equipRegion: 'tire',
    tier: 2,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 1,
      },
      {
        name: 'acceleration',
        value: 1,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1.1,
      },
      {
        name: 'grip',
        value: 0.8,
      },
      {
        name: 'gravity',
        value: 0.9,
      },
    ],
  },
  {
    id: 'j1j1j1',
    name: 'Transmission',
    vendorId: 'ccc',
    equipRegion: 'transmission',
    tier: 3,
    isEquipped: false,
    modifiers: [
      {
        name: 'maxSpeed',
        value: 0.7,
      },
      {
        name: 'acceleration',
        value: 1.5,
      },
      {
        name: 'steering',
        value: 1,
      },
      {
        name: 'breaking',
        value: 1,
      },
      {
        name: 'grip',
        value: 1,
      },
      {
        name: 'gravity',
        value: 1,
      },
    ],
  },
];

export const PLAYER = {
  login: 'lymak',
  nick: '$s$o$a60l$c82y$fa3m$c82a$a60k',
  inventory: {
    items: ITEMS,
  },
};

const PlayerResourceMock = {
  getPlayer() {
    return new Promise((resolve) => {
      const response = {
        data: PLAYER,
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  getInventory() {
    return new Promise((resolve) => {
      const response = {
        data: {
          items: ITEMS,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  equipItem() {
    return new Promise((resolve) => {
      const response = {
        data: {
          items: ITEMS,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },

  unequipItem() {
    return new Promise((resolve) => {
      const response = {
        data: ITEMS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default PlayerResourceMock;
