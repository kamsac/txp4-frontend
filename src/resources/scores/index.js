import axios from 'axios';
import dotenvConfiguration from '../../dotenv-configuration';

const ScoresResource = {
  getScores() {
    return axios.get(`${dotenvConfiguration.API_URL}/scores`);
  },
};

export default ScoresResource;
