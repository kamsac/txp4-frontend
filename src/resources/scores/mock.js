const SCORES = [
  {
    player: {
      login: 'lymak',
      nickname: 'lymak',
    },
    points: 4242,
  },
  {
    player: {
      login: 'kibes',
      nickname: 'kibes',
    },
    points: 3232,
  },
  {
    player: {
      login: 'tomek',
      nickname: 'tomek',
    },
    points: 2416,
  },
  {
    player: {
      login: 'omed',
      nickname: 'omed',
    },
    points: 2000,
  },
  {
    player: {
      login: 'tar',
      nickname: 'tar',
    },
    points: 1960,
  },
  {
    player: {
      login: 'oiz',
      nickname: 'oiz',
    },
    points: 680,
  },
  {
    player: {
      login: 'xetrow',
      nickname: 'xetrow',
    },
    points: 403,
  },
  {
    player: {
      login: 'ginok',
      nickname: 'ginok',
    },
    points: 30,
  },
  {
    player: {
      login: 'llif',
      nickname: 'llif',
    },
    points: 20,
  },
  {
    player: {
      login: 'ida',
      nickname: 'ida',
    },
    points: 10,
  },
];

const ScoresResourceMock = {
  getScores() {
    return new Promise((resolve, reject) => {
      const response = {
        data: SCORES,
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default ScoresResourceMock;