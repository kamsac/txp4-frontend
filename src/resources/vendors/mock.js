const VENDORS = [
  {
    id: 'aaa',
    name: "Super",
    color: "#8f0",
  },
  {
    id: 'bbb',
    name: "Safe",
    color: "#f80",
  },
  {
    id: 'ccc',
    name: "Heavy",
    color: "#800",
  },
  {
    id: 'ddd',
    name: "Light",
    color: "#08f",
  },
];

const VendorsResourceMock = {
  getVendors() {
    return new Promise((resolve, reject) => {
      const response = {
        data: VENDORS,
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default VendorsResourceMock;
