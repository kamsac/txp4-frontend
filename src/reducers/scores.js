import {RECEIVE_SCORES} from '../actions/scores';

const initialValue = [];

export function scoresReducer(state = initialValue, action) {
  switch(action.type) {
    case RECEIVE_SCORES:
      return [...action.payload.scores];
    default:
      return state;
  }
}
