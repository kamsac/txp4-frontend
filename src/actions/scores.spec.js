import moxios from 'moxios';
import { fetchScores, receiveScores } from './scores';
import { SCORES } from '../resources/scores/mock';
import dotenvConfiguration from '../dotenv-configuration';

test('receiveScores', () => {
  expect(receiveScores(SCORES)).toMatchSnapshot();
});

test('fetchScores', (done) => {
  moxios.install();
  dotenvConfiguration.API_URL = jest.fn().mockReturnValue('https://localhost:9000/api');

  const dispatchMock = jest.fn();
  moxios.withMock(() => {
    fetchScores()(dispatchMock);
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request
        .respondWith({
          status: 200,
          response: SCORES,
        })
        .then(() => {
          expect(request.url).toEqual(`${dotenvConfiguration.API_URL}/scores`);
          expect(dispatchMock).toBeCalledWith(receiveScores(SCORES));
          moxios.uninstall();
          done();
        });
    });
  });
});

test('fetchScores mock', (done) => {
  moxios.install();
  dotenvConfiguration.API_URL = jest.fn().mockReturnValue('');

  const dispatchMock = jest.fn();
  moxios.withMock(() => {
    fetchScores()(dispatchMock);
    moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request
        .respondWith({
          status: 200,
          response: SCORES,
        })
        .then(() => {
          expect(request.url).toEqual(`${dotenvConfiguration.API_URL}/scores`);
          expect(dispatchMock).toBeCalledWith(receiveScores(SCORES));
          moxios.uninstall();
          done();
        });
    });
  });
});
