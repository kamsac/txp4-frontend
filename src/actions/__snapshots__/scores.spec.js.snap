// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`receiveScores 1`] = `
Object {
  "payload": Object {
    "scores": Array [
      Object {
        "player": Object {
          "login": "lymak",
          "nick": "$s$o$a60l$c82y$fa3m$c82a$a60k",
        },
        "score": 4242,
      },
      Object {
        "player": Object {
          "login": "kibes",
          "nick": "$s$w$fffkibes",
        },
        "score": 3232,
      },
      Object {
        "player": Object {
          "login": "tomek",
          "nick": "$w$o$s$08ftom$fffek",
        },
        "score": 2416,
      },
      Object {
        "player": Object {
          "login": "omed",
          "nick": "$f80omed",
        },
        "score": 2000,
      },
      Object {
        "player": Object {
          "login": "tar",
          "nick": "$f00tar",
        },
        "score": 1960,
      },
      Object {
        "player": Object {
          "login": "oiz",
          "nick": "$o$w$000oiz",
        },
        "score": 680,
      },
      Object {
        "player": Object {
          "login": "xetrow",
          "nick": "$w$484xetrow",
        },
        "score": 403,
      },
      Object {
        "player": Object {
          "login": "ginok",
          "nick": "$n$880ginok",
        },
        "score": 30,
      },
      Object {
        "player": Object {
          "login": "llif",
          "nick": "$400$wllif",
        },
        "score": 20,
      },
      Object {
        "player": Object {
          "login": "ida",
          "nick": "$f00ida",
        },
        "score": 10,
      },
    ],
  },
  "type": "scores:fetchScoresSuccess",
}
`;
